includes/admin/class-wc-rest-payments-webhook-controller.php

```diff
@@ -301,9 +301,18 @@ private function process_webhook_payment_intent_succeeded( $event_body ) {
 		// Get an updated set of order properties to avoid race conditions when the server sends the paid webhook before we've finished processing the original payment request.
 		$order->get_data_store()->read( $order );
 
-		if ( ! $order->has_status( [ 'processing', 'completed' ] ) ) {
-			$order->payment_complete();
+		if ( $order->has_status( [ 'processing', 'completed' ] ) ) {
+			return;
 		}
+
+		// prevent parallel order processing attempts.
```

Comment:
```suggestion
		// Prevent parallel order processing attempts.
```

Tiniest of nitpicks here.

Reply:
Updated in https://github.com/Automattic/woocommerce-payments/pull/3520/commits/caed1c36cabe9c5b2dc7b351ac9a317602ccd12a. Thanks!